internal/pkg/manifest/validate.go

```diff
@@ -850,20 +850,27 @@ func (a AdvancedCount) Validate() error {
 		}
 	}
 
+	// Validate combinations with cooldown
+	if !a.Cooldown.IsEmpty() {
```

Comment:
Is this manifest valid?
```
count:
  range: 1-3
  cooldown:   # Optional.
    in: 3m        # Defaults to 2m
    out: 1m30s    # Defaults to 1m
  cpu_percentage: 
    value: 40   
    cooldown:
       in: 3m    
```
where cooldown is specified in both levels?

Reply:
Both the above manifests are valid, and the expected output should be that the lower level overwrites the upper level. I'm slightly uncertain about how it's done in code because as I'm writing the next part of the feature in `deploy/cloudformation/stack/transformers.go` the logic here is dependent on the order the lines are executed, but I don't know if there's a different way (best practices) that avoids this.
`var (
		CPUCooldown *template.Cooldown
		MemCooldown *template.Cooldown
	)

	// Write from generalized cooldown first
	generalCooldown := convertCooldown(a.Cooldown)
	CPUCooldown = generalCooldown
	MemCooldown = generalCooldown

	// If both generalized cooldown and specific cooldowns are specified, cooldown is overwritten by lower level config
	if cooldown := convertCooldown(a.CPU.ScalingConfig.Cooldown); cooldown != nil {
		CPUCooldown = cooldown
	}
	if cooldown := convertCooldown(a.Memory.ScalingConfig.Cooldown); cooldown != nil {
		MemCooldown = cooldown
	}

	autoscalingOpts.CPUCooldown = CPUCooldown
	autoscalingOpts.MemCooldown = MemCooldown`
	
	You can see in the above conversation that iamhopaul123 specified they should not be mutually exclusive:
	https://github.com/aws/copilot-cli/pull/3629#discussion_r890361447